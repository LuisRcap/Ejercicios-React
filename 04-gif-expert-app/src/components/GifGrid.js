import React from 'react'

const GifGrid = ({ category }) => {

    const getGifs = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=Pozh5uZofEN9CXolqKGTTcSXfGXdytxX';
        
        const resp = await fetch( url );
        const {data} = await resp.json();

        const gifs = data.map( (img) => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            };
        });

        console.log(gifs);
    };

    getGifs();

    return (
        <>
            <h3>{ category }</h3>
        </>
    );
};

export default GifGrid;