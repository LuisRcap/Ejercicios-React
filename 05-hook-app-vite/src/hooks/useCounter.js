import { useState } from "react"

export const useCounter = ( initialValue = 10 ) => {
    
    const [counter, setCounter] = useState(initialValue);

    const increment = (increase = 1) => {
        setCounter(counter + increase);
    };

    const decrement = (decrease = 1) => {
        if ( counter === 0 ) return;
        setCounter(counter - decrease);
    };

    const reset = () => {
        setCounter(initialValue);
    };

    return {
        counter,
        increment,
        decrement,
        reset
    }
}