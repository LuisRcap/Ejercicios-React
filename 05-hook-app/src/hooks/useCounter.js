import { useState } from "react"


export const useCounter = ( inicialState = 10 ) => {
    
    const [state, setState] = useState( inicialState ); // 10

    const increment = ( factor = 1 ) => {
        setState( state + factor );
    };

    const decrement = ( factor = 1 ) => {
        setState( state - factor );
    };

    const reset = () => {
        setState( inicialState );
    };

    return {
        state,
        increment,
        reset,
        decrement
    };

}
