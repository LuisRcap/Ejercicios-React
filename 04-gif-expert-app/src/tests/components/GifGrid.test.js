import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {
    
    const value = 'Star Wars';

    test('Debe mostrar el componente correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={ value } />);

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localgost/algo.jpg',
            title: 'Algo'
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={ value } />);

        expect( wrapper ).toMatchSnapshot();

    });
    
    
});
