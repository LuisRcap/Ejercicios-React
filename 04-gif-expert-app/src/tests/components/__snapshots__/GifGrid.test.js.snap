// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Star Wars
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Star Wars
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Algo"
      url="https://localgost/algo.jpg"
    />
  </div>
</Fragment>
`;
